v7/_posts/odata features/2017-12-26-06-00-JsonBatch.md

```diff
@@ -0,0 +1,330 @@
+		ODataMessageWriterSettings settings = new ODataMessageWriterSettings();
+	
+		using (ODataMessageWriter messageWriter = new ODataMessageWriter(requestMessage, settings))
+		{
+			ODataBatchWriter batchWriter = messageWriter.CreateODataBatchWriter();
+			batchWriter.WriteStartBatch();
+			string createRequestId = "contentId-1";
+	                
+			batchWriter.WriteStartChangeset();
+	                
+			ODataBatchOperationRequestMessage createOperationMessage = batchWriter.CreateOperationRequestMessage(
+				"POST",
+				new Uri(string.Format(CultureInfo.InvariantCulture, "{0}/{1}", serviceDocumentUri, "MySingleton")),
+				createRequestId);
+			using (ODataMessageWriter operationMessageWriter = new ODataMessageWriter(createOperationMessage))
+			{
+				ODataWriter entryWriter = operationMessageWriter.CreateODataResourceWriter();
+				ODataResource entry = new ODataResource();
+	                    
+				// skipped customized data population for initialization here... 
+	                    
+				entryWriter.WriteStart(entry);
+				entryWriter.WriteEnd();
+			}
+	
+			// A PATCH operation that depends on the preceding POST operation.
+			List<string> dependsOnIds = new List<string> { createRequestId };
+			ODataBatchOperationRequestMessage updateOperationMessage = batchWriter.CreateOperationRequestMessage(
+				"PATCH",
+				new Uri(string.Format(CultureInfo.InvariantCulture, "{0}/{1}", serviceDocumentUri, "$" + createRequestId)),
+				"updateReuqestId-1",
+				BatchPayloadUriOption.AbsoluteUri,
+				dependsOnIds);
+			using (ODataMessageWriter operationMessageWriter = new ODataMessageWriter(updateOperationMessage))
+			{
+				var entryWriter = operationMessageWriter.CreateODataResourceWriter();
+				var entry = new ODataResource();
+	                    
+				// skipped customized data population for update here... 
+	                    
+				entryWriter.WriteStart(entry);
+				entryWriter.WriteEnd();
+			}
+	
+			batchWriter.WriteEndChangeset();
+	
+			ODataBatchOperationRequestMessage queryOperationMessage = batchWriter.CreateOperationRequestMessage(
+				"GET",
+				new Uri(string.Format(CultureInfo.InvariantCulture, "{0}/{1}", serviceDocumentUri, "MySingleton")),
+				"readOperationId-1");
+			queryOperationMessage.SetHeader("Accept", "application/json;odata.metadata=full");
+	
+			batchWriter.WriteEndBatch();
+		}
+		// stream contains the JSON batch request payload.
+
+{% endhighlight %}
+
+Note that in JSON batch, `dependsOnIds` needs to include the request Id being referenced in the request's URL, as required by the JSON batch semantics.
+
+## Typical JSON Batch Request/Response Processing by an OData Service
+The processing of both JSON and Multipart/Mixed formats share the same common pattern driven by different content-type values.
+Starting from the following objects of OData types typically available in the OData service's processing of incoming batch request:
+
+{% highlight csharp %}
```

Comment:
use "~~~csharp"  instead of {% highlight csharp %}

Reply:
Updated.